src/codegen/utilities/interpreter_helpers.py

```diff
@@ -62,22 +62,25 @@ def get_interpreter_functions(metadata):
 def generate_interpreter_function_call_args(function_metadata):
     """Gets function call arguments."""
     function_call_args = []
-    for param in function_metadata.base_parameters:
+    size_value = []
+    size_param_index = None
+    for param in function_metadata.interpreter_parameters:
+        ## Deleting repeated size parameters
+        if param.has_explicit_buffer_size:
+            if size_value == param.size.value:
+                del function_call_args[size_param_index]
+
         if param.direction == "in":
             function_call_args.append(param.parameter_name)
             if param.has_explicit_buffer_size:
                 function_call_args.append(f"len({param.parameter_name})")
+                size_value = param.size.value
+                size_param_index = len(function_call_args) - 1
         else:
             if param.has_explicit_buffer_size:
+                function_call_args.append(param.parameter_name)
                 if param.size.mechanism == "ivi-dance":
```

Comment:
I think deleting the repeated parameter makes this function hard to follow. It would be clearer if the function didn't add a duplicate in the first place. 

Initializing `size_value` to a list and comparing it to a string doesn't help matters.

```suggestion
    params_to_skip = []
    for param in function_metadata.interpreter_parameters:
        if param.parameter_name in params_to_skip:
            continue
        if param.direction == "in":
            function_call_args.append(param.parameter_name)
            if param.has_explicit_buffer_size:
                function_call_args.append(f"len({param.parameter_name})")
                params_to_skip.append(param.size.value)
        else:
            if param.has_explicit_buffer_size:
                function_call_args.append(param.parameter_name)
                if param.size.mechanism == "ivi-dance":
```

BTW, this still assumes the array is immediately followed by the array size. If this assumption is valid, that's okay. Avoiding this assumption would be more complicated: add another loop to pre-scan for array parameters and change the main loop to add `len(array)` when processing the size parameter, not when processing the array parameter.

Reply:
Thanks for explaining.

Adding `len(array)` immediately after the array makes an assumption about the parameter order. This assumption is not valid in the case where multiple arrays share the same size parameter.

Can we eliminate that assumption and generate the size parameter in the right place?
```
    size_values = {}
    for param in function_metadata.interpreter_parameters:
        if param.direction == "in" and param.has_explicit_buffer_size:
            size_values.setdefault(param.size.value, f"len({param.parameter_name})")
        elif param.direction == "out" and param.has_explicit_buffer_size and param.size.mechanism == "ivi-dance":
            size_values.setdefault(param.size.value, "temp_size")

    for param in function_metadata.interpreter_parameters:
        if param.parameter_name in size_values:
            function_call_args.append(size_values[param.parameter_name])
        elif param.direction == "in":
            function_call_args.append(param.parameter_name)
        else:
```

If this won't work either, then:
- Update the comment to document the assumption you are making and the case where the assumption is not valid.
- Initialize size_value to the correct data type.

Reply:
Thanks for the logic suggestion, I used a similar implementation with some slight modifications to achieve the expected behavior.

Reply:
That still appends the size parameter while iterating over the array parameter, which is a big assumption.

My suggestion eliminates this assumption by appending the size parameter while iterating over the size parameter. This ensures that each parameter is appended in the right order, no matter what.

I tried it and it works. Here it is with your style changes:
```
def generate_interpreter_function_call_args(function_metadata):
    """Gets function call arguments."""
    function_call_args = []
    size_values = {}
    for param in function_metadata.interpreter_parameters:
        if param.has_explicit_buffer_size:
            if param.direction == "in":
                size_values[param.size.value] = f"len({param.parameter_name})"
            elif param.direction == "out":
                if param.size.mechanism == "ivi-dance":
                    size_values[param.size.value] = "temp_size"

    for param in function_metadata.interpreter_parameters:
        if param.parameter_name in size_values:
            function_call_args.append(size_values[param.parameter_name])
        elif param.direction == "in":
            function_call_args.append(param.parameter_name)
        elif param.direction == "out":
            if param.has_explicit_buffer_size:
                function_call_args.append(param.parameter_name)
            else:
                function_call_args.append(f"ctypes.byref({param.parameter_name})")
    return function_call_args
```

Reply:
Here in the logic the keys in the `size_value` dictionary are representing the `parameter.size.value`, but when accessing the dictionary the logic tries to access it through param.parameter_name and hence the `if param.parameter_name in size_values` is never true and none of the size params are getting added in my case.

Reply:
Thanks for the clarification, I have added a code comment as part of this PR to explain my assumptions, I will take up this logic in a separate PR as it requires some refactoring in the parameter filtering, we do now. Currently we filter out all the size value parameters which run based on `ivi-dance` mechanism and also any parameter with the name `size`.
This logic has to reverted in order for the suggested implementation to completely work for all functions, hence I am planning to take this up in a separate PR.

Reply:
Okay.